main.tf

```diff
@@ -248,88 +224,80 @@ resource "google_redis_instance" "gitlab" {
+  project_id       = var.project_id
+  location         = var.region
+  storage_class    = var.gcs_bucket_storage_class
+  prefix           = var.project_id
+  randomize_suffix = var.gcs_bucket_random_suffix
+  names = ["-gitlab-artifacts-",
+    "-gitlab-backups-",
+    "-gitlab-dependency-proxy-",
+    "-gitlab-external-diffs-",
+    "-gitlab-git-lfs-",
+    "-gitlab-packages-",
+    "-gitlab-registry-",
+    "-gitlab-pseudo-",
+    "-gitlab-runner-cache-",
+    "-gitlab-terraform-state-",
+    "-gitlab-tmp-backups-",
+    "-gitlab-uploads-"
+  ]
+  versioning = {
+    "-gitlab-artifacts-"        = var.gcs_bucket_versioning
+    "-gitlab-backups-"          = var.gcs_bucket_versioning
+    "-gitlab-dependency-proxy-" = var.gcs_bucket_versioning
+    "-gitlab-external-diffs-"   = var.gcs_bucket_versioning
+    "-gitlab-git-lfs-"          = var.gcs_bucket_versioning
+    "-gitlab-packages-"         = var.gcs_bucket_versioning
+    "-gitlab-registry-"         = var.gcs_bucket_versioning
+    "-gitlab-pseudo-"           = var.gcs_bucket_versioning
+    "-gitlab-runner-cache-"     = var.gcs_bucket_versioning
+    "-gitlab-terraform-state-"  = var.gcs_bucket_versioning
+    "-gitlab-tmp-backups-"      = var.gcs_bucket_versioning
+    "-gitlab-uploads-"          = var.gcs_bucket_versioning
+  }
+  force_destroy = {
+    "-gitlab-artifacts-"        = var.gcs_bucket_allow_force_destroy
+    "-gitlab-backups-"          = var.gcs_bucket_allow_force_destroy
+    "-gitlab-dependency-proxy-" = var.gcs_bucket_allow_force_destroy
+    "-gitlab-external-diffs-"   = var.gcs_bucket_allow_force_destroy
+    "-gitlab-git-lfs-"          = var.gcs_bucket_allow_force_destroy
+    "-gitlab-packages-"         = var.gcs_bucket_allow_force_destroy
+    "-gitlab-registry-"         = var.gcs_bucket_allow_force_destroy
+    "-gitlab-pseudo-"           = var.gcs_bucket_allow_force_destroy
+    "-gitlab-runner-cache-"     = var.gcs_bucket_allow_force_destroy
+    "-gitlab-terraform-state-"  = var.gcs_bucket_allow_force_destroy
+    "-gitlab-tmp-backups-"      = var.gcs_bucket_allow_force_destroy
+    "-gitlab-uploads-"          = var.gcs_bucket_allow_force_destroy
+  }
 
-resource "google_storage_bucket" "gitlab_external_diffs" {
-  name          = local.gitlab_external_diffs_bucket_name
-  location      = var.region
-  storage_class = var.gcs_bucket_storage_class
-  force_destroy = var.allow_force_destroy
+  bucket_lifecycle_rules = {
+    "-gitlab-backups-" = [{
+      action = {
+        type          = "SetStorageClass"
+        storage_class = "COLDLINE"
+      }
+      condition = {
+        age                   = var.gcs_bucket_backup_sc_change
+        matches_storage_class = var.gcs_bucket_storage_class
+      }
+      action = {
+        type = "Delete"
+      }
+      condition = {
+        age                   = var.gcs_bucket_backup_duration
+        with_state            = "ANY"
+        matches_storage_class = "COLDLINE"
+      }
+    }]
+  }
```

Comment:
i am not very sure about this policy, here we are basically moving to the COLD LINE storage, which means that you pay in advance for 90 days (at a lower price), so we should delete at least after 90 days of changing storage class, so we should use the number of backups to keep instead of tweaking the days that can lead to burning money.